src/routes/mods/[slug]/+page.svelte

```diff
@@ -0,0 +1,194 @@
+<script>
+	import Tag from '$lib/components/text/Tag.svelte';
+    import Check from '$lib/icon/Check.svelte';
+    import SvelteMarkdown from 'svelte-markdown';
+
+    import FullLoader from '$lib/components/FullLoader.svelte';
+    import Link from '$lib/components/text/Link.svelte';
+
+	import { goto } from '$app/navigation';
+	import { page } from '$app/stores';
+	let slug = $page.params.slug;
+
+    import { selectedVersion } from '$lib/stores';
+    import { getPackageInfo, isInstalledForVersion, getDeplist } from '$lib/api/contentdb';
+    import { downloadAndUnzip } from '$lib/api/download';
+    //import { openGame } from '$lib/shell';
+
+    import { onMount } from 'svelte';
+  	import { writable } from 'svelte/store';
+	
+    let packageInfo = false;
+    let isInstalled = false;
+    let isSupported = false; //TODO: figure out a way to do this...
+
+	let installing = false;
+	let currentinstalling = writable("");
+	let progress = writable(0);
+	let modstoinstall = [];
+	
+    let [ author, pack ] = slug.split('@');
+
+    onMount(updatePackage);
+	page.subscribe(updatePackage);
+
+	async function updatePackage() {
+		slug = $page.params.slug;
+		let slugParts = slug.split('@');
+        packageInfo = await getPackageInfo(slugParts[0], slugParts[1]);
+		//isInstalled = await isInstalledForVersion(slugParts[1], 'mods', $selectedVersion.name);
+	};
+
+    selectedVersion.subscribe(async(val) => {
+        if (!val.installed) {
+			isSupported = false;
+			return;
+		}
+
+        isInstalled = await isInstalledForVersion(pack, 'mods', val.name);
+		for (let i in packageInfo.releasesdata){
+			let release = packageInfo.releasesdata[i];
+			if (release.min_minetest_version){
+				isSupported = release.min_minetest_version.protocol_version >= val.branch.protocolVersion;
+				if (isSupported && release.max_minetest_version) {
+					isSupported = release.max_minetest_version.protocol_version <= val.branch.protocolVersion;
+				}
+			} else if (release.max_minetest_version) {
+				isSupported = release.max_minetest_version.protocol_version <= val.branch.protocolVersion;
+			}
+			if (isSupported) break;
+		}
+	});
+
+    async function install(version = '5.6.0') {
+		if (installing) return;
+		modstoinstall = await getDeplist(packageInfo, version);
+		modstoinstall.push(packageInfo);
+		
+        installing = true;
+		progress.set(0);
+		let installedprogress = 0;
+		for (let i in modstoinstall) {
+			let installpackage = modstoinstall[i];
+			progress.update(n => n + 1);
+			currentinstalling.set(installpackage.name);
+			try {
+				await downloadAndUnzip(installpackage.url, `/versions/${version}/mods/${installpackage.name}`);
+				installedprogress += 1;
+			}
+			catch (err) {
+				console.log(err);
+			}
+		}
+        installing = false;
+		isInstalled = installedprogress == modstoinstall.length;
+		return isInstalled;
+	}
+
+	async function toggleMod(pack, version = '5.6.0'){
```

Comment:
`toggleMod` is something that we would abstract to a dedicated lib file

Reply:
It is not yet done or working at all, so atm it doesn't matter, but yeah we can do that, but idk where to put it